### ISSUE TITLE
[iOS] audio stops playing in background

### ISSUE THREAD

**Issue by the4thfloor** [rxlabz/audioplayer#14]
I tried the example app on a real iPhone device. When I switch the phone off audio stops playing. When I switch it on again it continue playing.

Do you have any hints?

I added `audio` to the `UIBackgroundModes` in the `example/ios/Runner/Info.plist` but it doesn't fixed the issue.

**Comment by denvist** [comment]
Wow, is it true? Because background playing is a critical for my application.

**Comment by the4thfloor** [comment]
yes, the `mediaplayer.prepare` method blocks the ui thread, if you have a hugh file the prepare methods needs a couple of seconds.

**Comment by ficiverson** [comment]
I managed yo solve the issue with the background mode with this code

`do {
            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: .mixWithOthers)
            try AVAudioSession.sharedInstance().setActive(true)
        } catch {
            print(error)
        }`

**Comment by omarrida** [comment]
@ficiverson I'm about to start a Flutter project and playing audio in the background is a core requirement. Can you please confirm that you were able to handle it with your workaround?

**Comment by ficiverson** [comment]
@omarrida For me its working on a radiostation app (only iOS version tested). Now I can listen to streaming music on background but I dont know if it works with local files because I dont have that requirement

**Comment by eddywm** [comment]
The way to fix that is
1. Set  `AVAudioSession `category to backgrdound mode, I proposed a fix waiting to be merged : pull request #43 
2. Add ausio in` UIBackgroundModes` in your `.plist files.`

**Comment by ericduffett** [comment]
I'm still running into this issue on iOS in two situations:
1) I have a timer running, that when it completes, I want to play audio. It works with the screen unlocked, but not with the screen locked.
2) I am playing two audio files back to back. When the first one completes, the second one should start. It does not.

I have done the following:
1) Toggled the switch under capabilities - Targets - Runner - Background Modes - Audio, Airplay, and Picture in Picture.
2) Added this line to my AppDelegate: 

```
let audioSession = AVAudioSession.sharedInstance()
    do {
        try audioSession.setCategory(AVAudioSessionCategoryPlayback)
        print("set successfully")
    } catch {
        print("Setting category to AVAudioSessionCategoryPlayback failed.")
    }
```

3) As suggested here: https://stackoverflow.com/questions/54646498/keep-music-app-alive-in-background-ios-and-android, I added this line to my AppDelegate:
`UIApplication.shared.beginReceivingRemoteControlEvents()`

The audio will continue playing when I hit the lock button, but it will not trigger new audio to play while the phone is locked.

**Comment by gateshibill** [comment]
> The way to fix that is
> 
> 1. Set  `AVAudioSession `category to playback mode, I proposed a fix waiting to be merged : pull request [Audio output for speakers now supported on iOS #43](https://github.com/rxlabz/audioplayer/pull/43)
> 2. Add ausio in` UIBackgroundModes` in your `.plist files.`

I done：
1.AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback)
2.  <key>UIBackgroundModes</key>
        <array>
                <string>audio</string>
 </array>

 It still can't play in the background.

**Comment by gateshibill** [comment]
> The way to fix that is
> 
> 1. Set  `AVAudioSession `category to playback mode, I proposed a fix waiting to be merged : pull request [Audio output for speakers now supported on iOS #43](https://github.com/rxlabz/audioplayer/pull/43)
> 2. Add ausio in` UIBackgroundModes` in your `.plist files.`

ERROR: CANNOT PLAY MUSIC IN BACKGROUND. Message from code: "Error Domain=NSOSStatusErrorDomain Code=2003329396 "(null)""

**Comment by gateshibill** [comment]
> > The way to fix that is
> > 
> > 1. Set  `AVAudioSession `category to playback mode, I proposed a fix waiting to be merged : pull request [Audio output for speakers now supported on iOS #43](https://github.com/rxlabz/audioplayer/pull/43)
> > 2. Add ausio in` UIBackgroundModes` in your `.plist files.`
> 
> I done： 1.AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback) 2. UIBackgroundModes audio
> 
> It still can't play in the background.

fixed it, it work well
 try AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback, options: AVAudioSession.CategoryOptions.mixWithOthers)
                   NSLog("Playback OK")
                   try AVAudioSession.sharedInstance().setActive(true)
                   NSLog("Session is Active")
               } catch {
                   NSLog("ERROR: CANNOT PLAY MUSIC IN BACKGROUND. Message from code: \"\(error)\"")
               }
